Quality_Declaration_fastrtps.md

```diff
@@ -0,0 +1,137 @@
+
+13.  *Must have documentation for each item in the public API (e.g. functions, classes, etc.)*
+    
+The public API is documented through their generated [Doxygen documentation](http://www.eprosima.com/docs/fast-rtps/1.5.0/html/group___f_a_s_t_r_t_p_s___g_e_n_e_r_a_l___a_p_i.html) and also in the [User Manual](https://github.com/eProsima/Fast-RTPS/blob/master/doc/pdf/Fast%20RTPS%20-%20User%20Manual.odt) document provided in their repository.
+
+14.  *Must have a declared license or set of licenses*
+
+The declared License for the code in the repository is [Apache License 2.0](https://github.com/eProsima/Fast-RTPS/blob/master/LICENSE).
+
+15.  *Must have a copyright statement in each source file*
+
+A check to the source files on their repository shows that the source files include copyright statements to *“Proyectos y Sistemas de Mantenimiento SL (eProsima)”*. It is not stated if this is enforced with any kind of linter analysis.
+
+16.  *Must have a "quality declaration" document, which declares the quality level and justifies how the package meets each of the requirements*
+    
+This document will be the quality declaration supporting that `fastrtps` can be qualified as Quality Level 1 to be used within the ROS2 ecosystem.
+
+### Testing:
+
+17.  *Must have system tests which cover all items in the "feature" documentation*
+    
+There are several [tests](https://github.com/eProsima/Fast-RTPS/tree/master/test) in the `fastrtps` repository, the unit tests folder and the communication folders seem to be testing all the features of the software.
+
+18.  *Must have system, integration, and/or unit tests which cover all of the public API*
+    
+There are several [tests](https://github.com/eProsima/Fast-RTPS/tree/master/test) in the `fastrtps` repository, the unit tests folder, the communication, the profiling, and the realtime folders seem to be testing all the public API of the software.
+
+19.  *Must have code coverage, and a policy for changes*
+    
+Checking their [CI jobs](http://jenkins.eprosima.com:8080/), and the testing folders, there seems to be no automatic code coverage analysis being made or anything related to code coverage analysis.
+
+20.  *Performance tests, and performance regression policy*
+    
+Checking their [tests folder](https://github.com/eProsima/Fast-RTPS/tree/master/test), it is available a folder with performance tests, and also some [CI jobs](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Plots/) related to performance analysis. It is not possible to know whether there is a performance regression policy with the information from their repository.
+
+21.  *Linters and Static Analysis*
+    
+Based on the [CI jobs test results](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/lastSuccessfulBuild/testReport/), there seems there is no automatic static analysis being made on the code developed.
+
+
+### Dependencies:
```

Comment:
This section could be cleaned up a bit. What are we doing about non-ROS dependencies of non-ROS dependencies? It seems like we could follow the rabbit hole a long ways in that case...

Reply:
I added the suggestion to include testing for the package. After testing is added, might be considered making an argument over these external dependencies I think (with the argument of being downloaded from upstream + testing on the fast-rtps library itself)